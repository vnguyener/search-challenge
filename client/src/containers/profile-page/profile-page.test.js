import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { configure } from 'enzyme';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Adapter from 'enzyme-adapter-react-16';
import ProfilePage from '.';

configure({ adapter: new Adapter() });
const mockStore = configureMockStore([thunk]);

describe('profile page container', () => {
  it('renders correctly', () => {
    const store = mockStore({
      profiles: {
        selectedProfile: {
          photoUrl: 'https://placeimg.com/200/200/people?id=1',
          handle: 'dave',
          location: 'dallas',
          age: 24,
          photoCount: 0,
          id: 123,
        },
      },
    });
    const history = createMemoryHistory();
    history.push('/profiles/123');

    const wrapper = renderer
      .create(
        <Provider store={store}>
          <Router history={history}>
            <ProfilePage />
          </Router>
        </Provider>
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
