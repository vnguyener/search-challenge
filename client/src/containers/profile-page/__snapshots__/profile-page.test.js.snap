// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile page container renders correctly 1`] = `
<main
  className="profile-container"
>
  <div
    className="back-button clickable"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      keyboard_arrow_left
    </span>
     
    <p>
      Back
    </p>
  </div>
  <div
    className="card-container"
  >
    <img
      alt="Profile Main"
      className="resp-media"
      src="https://placeimg.com/200/200/people?id=1"
    />
    <div
      className="header-context-container"
    >
      <div
        className="description"
      >
        <p
          className="label"
        >
          24
           · 
          dallas
        </p>
      </div>
    </div>
    <div
      className="body-context-container"
    >
      <h2
        className="body-header"
      >
        <p
          className="link"
        >
          dave
        </p>
      </h2>
    </div>
  </div>
</main>
`;
